app/src/main/java/com/github/drawyourpath/bootcamp/database/MockDataBase.kt

```diff
@@ -0,0 +1,73 @@
+package com.github.drawyourpath.bootcamp.database
+
+import android.graphics.Color
+import android.widget.TextView
+import java.time.LocalDate
+import java.util.*
+import kotlin.collections.ArrayList
+import kotlin.collections.HashMap
+
+class MockDataBase : Database() {
+    var usersDataBase: ArrayList<String> = ArrayList<String>()
+    var usernameToPersonalInfoDataBase: HashMap<String, PersonalInfo> = HashMap()
+    var usernameToUserGoalsDataBase: HashMap<String, UserGoals> = HashMap()
+
+    override fun isUserNameAvailable(userName: String, outputText: TextView): Boolean {
+        //add an element in the user database to test the UI
+        usersDataBase.add("albert")
+        val unAvailableOutput: String = buildString {
+            append("*The username ")
+            append(userName)
+            append(" is NOT available !")
+        }
+        val availableOutput: String = buildString {
+            append("*The username ")
+            append(userName)
+            append(" is available !")
+        }
+        if (userName == "") {
+            outputText.text = buildString { append("The username can't be empty !") }
+            outputText.setTextColor(Color.RED)
+            return false
+        } else if (usersDataBase.contains(userName)) {
+            outputText.text = unAvailableOutput
+            outputText.setTextColor(Color.RED)
+            return false
+        } else {
+            outputText.text = availableOutput
+            outputText.setTextColor(Color.GREEN)
+            return true
+        }
+    }
+
+    override fun setUserName(userName: String, outputText: TextView): Boolean {
+        if (isUserNameAvailable(userName, outputText)) {
+            usersDataBase.add(userName)
+            return true
+        }
+        return false
+    }
+
+    override fun setPersonalInfo(username: String, firstname: String, surname: String, dateOfBirth: LocalDate) {
+        val userPersonalInfo: PersonalInfo = PersonalInfo(firstname, surname, dateOfBirth)
+        usernameToPersonalInfoDataBase.put(username, userPersonalInfo)
+    }
+
+    override fun setUserGoals(username: String, distanceGoal: Int, timeGoal: Int, nbOfPathsGoal: Int) {
+        val userGoals: UserGoals = UserGoals(username,distanceGoal,timeGoal, nbOfPathsGoal)
+        usernameToUserGoalsDataBase.put(username, userGoals)
+    }
+}
+
+//create some enum to facilitate the tests
+class PersonalInfo(firstname: String, surname: String, dateOfBirth: LocalDate) {
```

Comment:
You might want to use [Data Classes](https://kotlinlang.org/docs/data-classes.html) maybe.

Reply:
Thanks for this informations, but the name in the Data Classes don't separate the first name and the surname.